nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-5-processors/src/main/java/org/apache/nifi/processors/elasticsearch/JsonQueryElasticsearch5.java

```diff
@@ -0,0 +1,291 @@
+@CapabilityDescription("A processor that allows the user to run a query (with aggregations) written with the " +
+        "ElasticSearch JSON DSL. It currently does not support pagination.")
+public class JsonQueryElasticsearch5 extends AbstractProcessor {
+    public static final Relationship REL_SUCCESS = new Relationship.Builder().name("success")
+            .description("All original flowfiles that don't cause an error to occur go to this relationship. " +
+                    "This applies even if you select the \"break up hits\" option to send individual hits to the " +
+                    "\"hits\" relationship.").build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder().name("failure")
+            .description("All FlowFiles that cannot be read from Elasticsearch are routed to this relationship").build();
+
+    public static final Relationship REL_HITS = new Relationship.Builder().name("hits")
+            .description("Search hits are routed to this relationship.")
+            .build();
+
+    public static final Relationship REL_AGGREGATIONS = new Relationship.Builder().name("aggregations")
+            .description("Aggregations are routed to this relationship.")
+            .build();
+
+    public static final PropertyDescriptor INDEX = new PropertyDescriptor.Builder()
+            .name("el5-fetch-index")
+            .displayName("Index")
+            .description("The name of the index to read from")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor TYPE = new PropertyDescriptor.Builder()
+            .name("el5-type")
+            .displayName("Type")
+            .description("The type of this document (used by Elasticsearch for indexing and searching)")
+            .defaultValue("")
+            .required(false)
+            .expressionLanguageSupported(true)
+            .addValidator(Validator.VALID)
+            .build();
+
+    public static final PropertyDescriptor QUERY = new PropertyDescriptor.Builder()
+            .name("el5-query")
+            .displayName("Query")
+            .description("A query in JSON syntax, not Lucene syntax.")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final AllowableValue BREAK_UP_YES = new AllowableValue(
+        "breakup-yes",
+        "Yes",
+        "Break up results."
+    );
+    public static final AllowableValue BREAK_UP_HITS_NO = new AllowableValue(
+        "breakup-no",
+        "No",
+        "Don't break up results."
+    );
+
+    public static final PropertyDescriptor BREAK_UP_HITS = new PropertyDescriptor.Builder()
+            .name("el5-break-up-hits")
+            .displayName("Break up search results")
```

Comment:
This is just a style comment (so not a requirement just a suggestion), other processors use the term "split" rather than "break up".

Reply:
Fair enough.